### ISSUE TITLE
Unit test with karma/mocha/chai/sinon

### ISSUE THREAD

**Issue by griest024** [stasson/vue-mdc-adapter#192]
Jest is fine, but there isn't much room for customization. I would like to set up unit testing with [karma](https://github.com/karma-runner/karma) as the runner, [mocha](https://github.com/mochajs/mocha) as the framework, [chai](https://github.com/chaijs/chai) as the assertion library, and [sinon](https://github.com/sinonjs/sinon) for mocks, spies, and stubs.

This set up would allow for much greater customization, though it does require more configuration. Fortunately, between each layer there are projects which take care of the integration for you.

Setting up unit testing in this way has a number of benefits including multi-broswer testing. With [karma launchers](https://www.npmjs.com/search?q=karma-launcher) you can automatically run tests on many different browsers. There's even a [karma saucelabs integration](https://github.com/karma-runner/karma-sauce-launcher) if you want to go that route. It would be very helpful in discovering specific browser issues (IE11).

@stasson Are you open to this? I can create a branch and PR it so you can play around with the setup. Ultimately after everything is set up, its no more difficult to use than jest.

**Comment by stasson** [comment]
@griest024 

I'm really happy with jest/jsdom/vue-test-utils today. The only difficult part I see now is mocking JSDOM with the level of DOM support we need for material-components ( media queries, MutationObserver, animationFrame ...) It may have been easier with PhantomJS, but I think the test polyfills I added are ok for unit testing... 

As of now, the unit test baseline only has sanity checks. I'd like to add code coverage and much more test.specs. unit tests should focus only on vue-mdc-adapter JS/DOM validity. meaning no point testing MDCWeb Library and no point running unit test in browsers. 

That being said we should also have end to end _browser_ tests to run component integration tests across browsers . For that I think we should use sauceLab & webdriver.io. Here we do not care about vue-mdc-adapter javascript internals, but want to verify components do properly behave and to check integration of MDCWeb is fine.

Bottom line: 

1. I want to keep jest/jsdom/vue-test-utils for _unit test_. As of today I'd rather put the efforts on adding many more test.spec and  code coverage... rather than changing the test runner.
1. We also need _browser & integration tests_ and I'm ok with investigating e2e test strategy and the stack ( SauceLab + webdriver.io + chimp.js???  + mocha/chai???)

**Comment by griest024** [comment]
@stasson yeah you make some good points. I will submit a PR that hopefully will provide the necessary environment for end to end testing.
